automation/src/test/java/framework/DriverManager.java

```diff
@@ -0,0 +1,36 @@
+package framework;
+
+import java.util.HashMap;
+
+import org.openqa.selenium.WebDriver;
+
+public abstract class DriverManager {
+	
+	private WebDriver driver;
+	
+	public abstract void createDriver();
+
+	protected void setDriver() {
+		this.driver = getDriver();
+	}
+	
+	public WebDriver getDriver() {
+		ConfigurationProvider cfgProvider = new ConfigurationProvider();

```

Comment:
This violates Single Responsibility Principle. The DriverManager should not need to read the config file.